android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java

```diff
@@ -423,9 +430,23 @@ private void removeParticipantVideo(VideoTrack videoTrack) {
         videoTrack.removeRenderer(primaryVideoView);
     }
 
+
+    private void selectSpeakerAsAudioOutput() {
+        List<AudioDevice> devices = audioSwitch.getAvailableAudioDevices();
+        for (AudioDevice audioDevice: devices) {
+            if(audioDevice instanceof AudioDevice.Speakerphone) {
+                audioSwitch.selectDevice(audioDevice);
+                break;
+            }
+        }
+    }
+
     private void connectToRoom(String roomName, String accessToken) {
         audioSwitch = new AudioSwitch(getApplicationContext());
 
+        AudioDevice selectedDevice = audioSwitch.getSelectedAudioDevice();
+        if (selectedDevice instanceof  AudioDevice.Earpiece) selectSpeakerAsAudioOutput();
```

Comment:
one space too much :D

Reply:
I think the space here makes sense, the above two lines belong to difference concept than below

Reply:
spacen, no line break, between `instanceof` and `AudioDevice.Earpiece`

Reply:
oh, sorry I misunderstood

Reply:
fixed